File: cifrado/escritorio/codigo/Kotlin/src/procesar/Descifrador.kt
package procesar

import alfabeto.Alfabeto
import kotlin.properties.Delegates

class Descifrador(alfabeto: Alfabeto){
    var alfabeto : Alfabeto by Delegates.notNull()

    init {
        this.alfabeto = alfabeto
    }

    fun descifrarCaracter(recorrido:Int,caracter:Char): Char {
        val posicion = alfabeto.obtenerPosicion(caracter)
        if (posicion > 0)
            return alfabeto.obtenerCaracter(posicion-recorrido)
        else
            return caracter
    }

    fun descifrarTexto(recorrido:Int,texto:String): String {
        var textoCifrado = ""
        texto.toCharArray().forEach {
            textoCifrado = textoCifrado + descifrarCaracter(recorrido,it)
        }

        return textoCifrado
    }
}

File: cifrado/escritorio/codigo/Ruby/procesar/Descifrador.rb
class Descifrador
    
    def initialize(alfabeto)
            @alfabeto = alfabeto
    end

    def descifrarCaracter(recorrido,caracter)
        posicion = @alfabeto.obtenerPosicion(caracter)
        if posicion > 0
            return @alfabeto.obtenerCaracter(posicion-recorrido)
        else
            return caracter
        end
    end

    def descifrarTexto(recorrido,texto)
        textoPlano = ""
        texto.chars.each{
            |caracter| textoPlano = textoPlano + descifrarCaracter(recorrido,caracter)
        }
        return textoPlano
    end
end